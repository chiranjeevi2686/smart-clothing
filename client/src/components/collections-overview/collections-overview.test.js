import React from "react";
import { render } from '@testing-library/react';
import CollectionsOverview from './collections-overview.component';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const defaultStore = {
    shop: {
        collections: {
            hats: {
                id: 1,
                items: [{
                    id: 1,
                    imageUrl: 'test-url',
                    name: 'test-name',
                    price: 10
                }],
                routeName: 'test',
                title: 'test-title'
            }
        },
        errorMessage: '',
        isFetching: false
    }
};
const mockStore = configureStore();
describe('Collection overview component', () => {
    it('should render collection overview', async () => {
        const { asFragment } = render(<Provider store={mockStore(defaultStore)}>
            <CollectionsOverview />
        </Provider>)
        expect(asFragment()).toMatchSnapshot();

    })
})
