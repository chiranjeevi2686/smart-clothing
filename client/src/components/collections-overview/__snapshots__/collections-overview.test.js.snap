// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collection overview component should render collection overview 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ hBPBCu"
  >
    <div
      class="sc-fubCfw kNQsBt"
    >
      <h1
        class="sc-pFZIQ bWsjRN"
      >
        TEST-TITLE
      </h1>
      <div
        class="sc-jrAGrp kXpyIe"
      >
        <div
          class="sc-dlfnbm kgpEqj"
        >
          <div
            class="sc-eCssSg kLJAJi"
          />
          <div
            class="sc-jSgupP hfwbgV"
          >
            <span
              class="sc-gKsewC gzFpBu"
            >
              test-name
            </span>
            <span
              class="sc-iBPRYJ kwiMLn"
            >
              $10
            </span>
          </div>
          <button
            class="sc-gsTCUz cvicRc sc-hKgILt cAydbA"
            data-test-id="add-cart-button"
          >
            ADD TO CART
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
